service_configuration_lib/spark_config.py

```diff
@@ -307,20 +309,23 @@ def get_dra_configs(spark_opts: Dict[str, str]) -> Dict[str, str]:
                 min_executors, int(int(spark_opts['spark.dynamicAllocation.maxExecutors']) *
                                    float(min_executor_ratio)),
             )
+        # set minExecutors equal to 0 for Jupyter Spark sessions
+        if 'jupyterhub' in spark_app_name:
```

Comment:
Add a log here in lines with: "Looks like you are launching spark session from jupyterhub notebook. Setting min executor count to 0, to save spark costs when any spark action is not running".

Reply:
Adding

Reply:
In future version, we can have a slightly stricter regex check "^jupyterhub_.*".
Make sure to create a ticket to track, you can do it after pushing.